LSL/liblsl/src/sample.h

```diff
@@ -111,6 +125,30 @@ namespace lsl {
 				return result;
 			}
 
+			/// garbage collection for unmanaged samples
+			std::vector<sample*> garbage_vec;
+			boost::thread *garbage_thread;
+			int garbage_cnt;
+			bool stop_garbage;
+			boost::mutex garbage_mutex;
+			void collect_garbage() {
+				int cnt;
+				while (!stop_garbage) {
+					cnt=0;
+					// trick to lock for shortest possible time
```

Comment:
I'm unfamiliar with this trick. Can you please elaborate? Why don't you have to lock when you're emptying out garbage_vec?

Reply:
This doesn't work for the reasons I write below, but I believe this 'trick' would work because the garbage collector is on the same thread as the sample creator. The memory in the vector is guaranteed (only it's not, ha ha ha!) to be old and no longer in play. Only after 1000 old samples are done with does the memory get de-allocated. At least, that was my reasoning. 

This is clunky as hell, and I'm embarassed to know that anyone is actually looking at it! I was just eager to get it working so I cut a lot of corners.

Reply:
No! You are totally right, Chadwick! This isn't safe at all. It wouldn't work even if it worked. The 'trick' in the comment is more like a hack.  I lock on the ref counter which I assigned to a local so that I could compare it and not have the lock around the sleeping part.